ql/src/java/org/apache/hadoop/hive/ql/parse/BaseSemanticAnalyzer.java

```diff
@@ -330,6 +332,17 @@ public void analyze(ASTNode ast, Context ctx) throws SemanticException {
     initCtx(ctx);
     init(true);
     analyzeInternal(ast);
+    postAnalyze();
+  }
+
+  /**
+   * This method is called after analyzeInternal, assuming all the semantic context is present (analyzeInternal is
+   * overridden by the subclasses of BaseSemanticAnalyzer).
+   * The reason why postAnalyze is private is that it's not supposed to be overridden.
+   */
+  private void postAnalyze() {
```

Comment:
why do we need 1 more `postAnalyze`, can't we move `setUsedTables` in `endAnalysis`?

Reply:
good catch! it works properly in endAnalysis too
handled in: https://github.com/apache/hive/pull/5613/commits/812bb17b47d6f6f10deb706158e3959b38ddaea2

Reply:
if the build fails, please set the `usedTables` first and then `queryType`

Reply:
failed :) let me do it

Reply:
THX